config/manager/manager.yaml

```diff
@@ -23,12 +23,22 @@ spec:
         control-plane: controller-manager
     spec:
       securityContext:
-        runAsNonRoot: true
+        runAsNonRoot: false
+      volumes:
```

Comment:
Add these via overlay patches so that non-peerpod users don't get them

Reply:
I'm not entirely sure I understand what you have in mind. Are you talking about changing make deploy based on some peerpod env variable to keep this out of manager.yaml?

Reply:
yes, that was my suggestion

Reply:
I have devised a way to do this but kustomize is exhibiting unexpected behavior at least in my opinion.
My patches go in and it all works but the config/default/kustomization.yaml file gets corrupted so that anything below the patches list gets deleted regardless of if I am doing a patch add or a patch delete. So all the WEBHOOK and CERTMANGER comments go away.

```
# Mount the controller config file for loading manager configurations
# through a ComponentConfig type
#- manager_config_patch.yaml

# To enable the peerpod controllers
- path: peerpods/peerpods-namespace.yaml
- path: peerpods/enable-peerpods.yaml
  target:
    group: apps
    kind: Deployment
    name: controller-manager
    version: v1

# [WEBHOOK] To enable webhook, uncomment all the sections with [WEBHOOK] prefix including the one in
# crd/kustomization.yaml
#- manager_webhook_patch.yaml

# [CERTMANAGER] To enable cert-manager, uncomment all sections with 'CERTMANAGER'.
# Uncomment 'CERTMANAGER' sections in crd/kustomization.yaml to enable the CA injection in the admission webhooks.
# 'CERTMANAGER' needs to be enabled to use ca injection
#- webhookcainjection_patch.yaml

```
```
Makefile 

deploy: manifests kustomize ## Deploy controller to the K8s cluster specified in ~/.kube/config.
        cd config/manager && $(KUSTOMIZE) edit set image controller=${IMG}
ifneq (, $(PEERPODS))
        @echo Peerpods will be deployed and added to the manager config yamls.
        cd config/default && $(KUSTOMIZE) edit add patch --kind Deployment --version  v1 --name controller-manager --group apps  --path peerpods/enable-peerpods.yaml
        cd config/default && $(KUSTOMIZE) edit add patch --path peerpods/peerpods-namespace.yaml
endif
        $(KUSTOMIZE) build config/default | kubectl apply -f -
```
The other alternative is to not make this dynamic patching and instead leave the patch paths in as part of the yaml and do what seems to be pretty standard which is comment them all out with the instructions

 [PEERPODS] To enable peerpods, uncomment all the sections with [PEERPODS]  etc, etc, etc

Reply:
I'm going to try the latest version of kustomize to see if that helps